src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -68,9 +67,20 @@ class ResourceService(
                 Resource(resourceTypeAdmin.name, ResourceId(rtName.value), Set.empty)
             }
           }
+          _ <- updateEffectivePolicyTables(samRequestContext, newOrUpdatedResourceTypeNames)
```

Comment:
I think we need to do it after `persistResource`. I tried adding it to the `upsertResourceTypes` method, and that failed my tests, so I think it needs to happen there. I am going to add a lock to it, but I think it should be okay to do it outside of the other lock, because things will catch up if changes are made between upsert and updating the effective policy tables.

Reply:
What failed in the tests? Persisting the `resource-type-admin` resources shouldn't impact our ability to update effective policy tables from what I can tell

Reply:
The permissions check at the end of the test failed, which means the insert query and/or the delete query returned a different response. I didn't dig into why, but I can.

Reply:
(This was the test failure. Looks like maybe the delete failed?)

true was not equal to false
ScalaTestFailureLocation: org.broadinstitute.dsde.workbench.sam.service.ResourceServiceSpec at (ResourceServiceSpec.scala:1132)
Expected :false
Actual   :true
<Click to see difference>